packages/foam-vscode/src/features/wikilink-reference-generation.ts

```diff
@@ -135,10 +135,10 @@ async function generateReferenceList(
 ): Promise<string[]> {
   const filePath = doc.fileName;
 
-  const id = dropExtension(basename(filePath));
+  const note = foam.getNoteByURI(filePath);
 
   const references = uniq(
-    createMarkdownReferences(foam, id, includeExtensions()).map(
+    createMarkdownReferences(foam, note, includeExtensions()).map(
```

Comment:
speaking of TS missing typing errors, if I am not mistaking `note` here is `Note | null` but the parameter needs to be `Note`.
I honestly don't believe it could ever be `null` in this code path, so I don't expect this to fail (but we should probably still check to be sure) - what I am concerned about is TS not reporting a straightforward type error